### ISSUE TITLE
IndexSizeError: cfi_navigation_logic.js - findVisibleLeafNodeCfi

### ISSUE THREAD

**Issue by bradleygore** [readium/readium-shared-js#314]
#### Expected Behaviour
Get the range of selected nodes and highlight them.

#### Observed behaviour
`findVisibleLeafNodeCfi` is generating this error:
**IndexSizeError: Index or size is negative or greater than the allowed amount**

#### Steps to reproduce
1. Open an epub with a fixed-layout and an image background
2. Set breakpoint at [highlights manager line 117](https://github.com/readium/readium-shared-js/blob/34cfd38956143261e58b23711cca3d1c69d76c26/plugins/highlights/manager.js#L117) and if `spineItem.paginationInfo` isn't defined, default it to `{}` (see [this bug](https://github.com/readium/readium-shared-js/issues/280) for reference)
3: Select some text
4: Try to add highlight

### Product
* Readium cloud reader app
  * latest development build uploaded at https://readium.firebaseapp.com
  * Firefox 48
  * Multiple

#### Additional information
I'm working on getting an example fixed-layout epub with image background to upload with this issue.

**Comment by danielweck** [comment]
Duplicated issue? https://github.com/readium/readium-shared-js/issues/280

**Comment by bradleygore** [comment]
No, this isn't a dup. This is the error that you'll see once #280 is fixed, as I've included the workaround for #280 in the instructions for reproducing the issue.

**Comment by danielweck** [comment]
Okay thanks for the clarification! :)

**Comment by danielweck** [comment]
@bradleygore I suspect that the empty `paginationInfo` object simply "postpones" the occurrence of the bug (because the data in `paginationInfo` is needed for highlights to function properly, right?). I need to run some tests too, but ultimately, `paginationInfo` cannot just be empty.

**Comment by bradleygore** [comment]
@danielweck You're very likely right - I'm still trying to grok what all is going on, so could be that once the `paginationInfo` issue is fixed the right way, vs my poor hack around, that this will get resolved along with it :)
